cpp/src/arrow/flight/sql/odbc/odbcabstraction/include/odbcabstraction/odbc_impl/encoding_utils.h

```diff
@@ -77,4 +77,37 @@ inline size_t ConvertToSqlWChar(const std::string& str, SQLWCHAR* buffer,
   }
 }
 
+/// @brief Convert buffer of SqlWchar to standard string
+/// @param wchar_msg[in] SqlWchar to convert
+/// @param msg_len[in] Number of characters in wchar_msg
+/// @return wchar_msg in std::string format
+inline std::string SqlWcharToString(SQLWCHAR* wchar_msg, SQLSMALLINT msg_len = SQL_NTS) {
+  if (wchar_msg == nullptr) {
+    return std::string();
+  }
+
+  // Size of SQLWCHAR depends on the platform
+  size_t wchar_size = sizeof(SQLWCHAR);
+
+  // Assert that size of SQLWCHAR and wchar_t are the same
+  assert(wchar_size == sizeof(wchar_t));
+
+  // Copy SQLWCHAR into a temp wstring
+  std::wstring wstring_msg;
+
+  if (msg_len != SQL_NTS) {
+    for (int i = 0; wchar_msg[i] != 0 && i < msg_len; i++) {
+      wstring_msg.push_back(wchar_msg[i]);
+    }
+  } else {
+    for (int i = 0; wchar_msg[i] != 0; i++) {
+      wstring_msg.push_back(wchar_msg[i]);
+    }
+  }
+
+  // Converter for SQLWCHAR to wstring
+  static std::wstring_convert<std::codecvt_utf8<wchar_t>> conv;
```

Comment:
You shouldn't use wchar_t here if you want to support multiple DMs. It should be based on the driver manager's unicode encoding (which can be a setting or be auto-detected) and use char16_t or char32_t for UTF-16 and UTF-8 respectively.

Reply:
Updated logic for `SqlWcharToString`